tests/test_execution.py

```diff
@@ -55,27 +49,18 @@ def test_tuples_execution_on_same_node(global_execution_env):
 
 
 @pytest.mark.slow
-def test_federated_learning_workflow(global_execution_env):
+def test_federated_learning_workflow(factory, client, default_datasets):
     """Test federated learning workflow on each node."""
-    factory, initial_assets, network = global_execution_env
-    client = network.clients[0]
 
     # create test environment
     spec = factory.create_algo()
     algo = client.add_algo(spec)
 
-    # get first dataset of each client
-    # because there is only one dataset created by client, we can get all the datasets
-    datasets = initial_assets.datasets
-
-    # check there is one dataset per node in the network
-    assert set([d.owner for d in datasets]) == set([c.node_id for c in network.clients])
-
     # create 1 traintuple per dataset and chain them
     traintuple = None
     rank = 0
     compute_plan_id = None
-    for dataset in datasets:
+    for dataset in default_datasets:
```

Comment:
Could you add a coment explaining that default_datasets does contain datasets on each node and that as a result we can use it for federated learning.